legate/core/runtime.py

```diff
@@ -346,6 +358,31 @@ def allocate_field(self, field_size: Any) -> tuple[Region, int, bool]:
         return self._region, field_id, revived
 
 
+def _try_reuse_field(
```

Comment:
I suggest just blocking on the first Future, I'm worried that two shards might test the same future at slightly different points in time, one of them finds it ready and the other doesn't, and they end up using different fields.

```
def _try_reuse_field(
     free_fields: Deque[tuple[Region, int, Union[Future, None]]]
) -> Optional[tuple[Region, int]]:
    if len(free_fields) == 0:
        return None
    field_info = free_fields.popleft()
    if field_info[2] is not None:
        field_info[2].wait()
    return field_info[0], field_info[1]
```

Reply:
makes sense